packages/ap-storage/docs/async_uploads.rst

```diff
@@ -0,0 +1,229 @@
+Async File Uploads
+==================
+
+Overview
+--------
+
+The Async Upload feature in alliance-platform-storage provides a seamless way to handle asynchronous file uploads in Django applications.
+In this context, the async in "Async Upload" means the file is immediately uploaded to the server (e.g. S3, Azure, local server) while the user
+continues to fill out a form. A reference (the key) to the uploaded file is stored in the form and handled in the Django model save when
+processing the form submission. See `How it works`_ for more specifics.
```

Comment:
This is great - it took me quite a while to originally figure out what the actual point of the async upload widget was, and this explains it very well.